TKGame/Input.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Input;
+using System.Linq;
+
+
+namespace TKGame
+{
+    static class Input
+    {
+        private static KeyboardState keyboardState, lastKeyboardState;
+        private static MouseState mouseState, lastMouseState;
+        private static GamePadState gamepadState, lastGamepadState;
+
+        private static bool isAimingWithMouse = false;
+
+        public static Vector2 MousePosition { get { return new Vector2(mouseState.X, mouseState.Y); } }
+
+        public static void Update()
+        {
+            // Maintain previous frame's state
+            lastKeyboardState = keyboardState;
+            lastMouseState = mouseState;
+            lastGamepadState = gamepadState;
+
+            // Update state to current frame's input
+            keyboardState = Keyboard.GetState();
+            mouseState = Mouse.GetState();
+            gamepadState = GamePad.GetState(PlayerIndex.One);
+
+            // Disable mouse aiming if GamePad Input detected
+            if (gamepadState.ThumbSticks.Left != Vector2.Zero)
+                isAimingWithMouse = false;
+            else if (MousePosition != new Vector2(lastMouseState.X, lastMouseState.Y))
+                isAimingWithMouse = true;
+        }
+
+        // Checks if ANY key was just pressed
+        public static bool WasKeyPressed(Keys key)
+        {
+            return lastKeyboardState.IsKeyUp(key) && keyboardState.IsKeyDown(key);
+        }
+
+        // Same but for GamePad
+        public static bool WasButtonPressed(Buttons button)
+        {
+            return lastGamepadState.IsButtonUp(button) && gamepadState.IsButtonDown(button);
+        }
+
+        public static Vector2 GetMovementDirection()
+        {
+            Vector2 direction = gamepadState.ThumbSticks.Left;
+
+            if (keyboardState.IsKeyDown(Keys.A))
+                direction.X -= 1;
+            if (keyboardState.IsKeyDown(Keys.D))
+                direction.X += 1;
+            if (keyboardState.IsKeyDown(Keys.W))
+                direction.Y -= 1;
+            if (keyboardState.IsKeyDown(Keys.S))
+                direction.Y += 1;
+
+            //Clamp Vector length to a max of 1
+            if (direction.LengthSquared() > 1)
```

Comment:
I don't necessarily see a reason to use the `LengthSquared` over just `Length`, since you're wanting to know if the length of the vector is greater than 1 anyways

Reply:
I would have to compare the two values. L^2 just makes it so negatives aren't an issue

Reply:
In that case you maybe could've done `Math.Abs(direction.Length) > 1`?

Reply:
Seems like just more computation for the same result